types/address.go

```diff
@@ -91,15 +94,18 @@ func (a *Address) Decode(decoder scale.Decoder) error {
 	return nil
 }
 
-func (a Address) Encode(encoder scale.Encoder) error {
+func (a Address) Encode(ctx context.Context, encoder scale.Encoder) error {
```

Comment:
Encoding is the main issue here, since decoding will work already.
If flag is passed in context then it will avoid adding the prefix that is required with indices pallet.